File: test_milestone1/repos/pure-python/src/greeter.py
"""Greeting utilities."""


def greet(name: str) -> str:
    """Greet someone by name."""
    message = format_greeting("Hello", name)
    print(message)
    return message


def format_greeting(template: str, name: str) -> str:
    """Format a greeting string."""
    return f"{template}, {name}!"


def farewell(name: str) -> str:
    """Say goodbye."""
    msg = f"Goodbye, {name}!"
    print(msg)
    return msg

File: test_milestone1/repos/pure-ts/src/greeter.ts
// Greeting utilities.

function greet(name: string): string {
    const message = formatGreeting("Hello", name);
    console.log(message);
    return message;
}

const formatGreeting = (template: string, name: string): string => {
    return template + ", " + name + "!";
};

const farewell = (name: string): string => {
    const msg = "Goodbye, " + name + "!";
    console.log(msg);
    return msg;
};
